src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala

```diff
@@ -31,125 +31,155 @@ import org.dia.tensors.AbstractTensor
-        extract_masked_data(destNode, lats, lons, tightestBox)
+  /**
+   * Writes the node to a netCDF file
+   *
+   * @param directory The directory to write the file to
+   * @param fileName  String The full path and name of the netcdfFile.
+   * @param varData   ucar.ma2.ArrayInt.D2  2D data to be written to the file.
+   * @param lats      Array[Double] of the latitudes to be used
+   * @param lons      Array[Double] of the longitudes to be used
+   */
+  def writeNodeToNetCDF(directory: String,
+                        fileName: String,
+                        varData: ucar.ma2.ArrayInt.D2,
+                        lats: Array[Double],
+                        lons: Array[Double]): Unit = {
+    val filepath = directory + fileName
+    try {
+      val fsplit = filepath.split("_")
+      val latMin = fsplit(1).toInt
+      val latMax = fsplit(2).toInt
+      val lonMin = fsplit(3).toInt
+      val lonMax = fsplit(4).dropRight(3).toInt
+      val lats1 = lats.slice(latMin, latMax + 1)
+      val lons1 = lons.slice(lonMin, lonMax + 1)
+      val datafile = NetcdfFileWriter.createNew(NetcdfFileWriter.Version.netcdf3, filepath, null)
+      // Create netCDF dimensions
+      val lonDim = datafile.addDimension(null, "longitudes", lons1.length)
+      val latDim = datafile.addDimension(null, "latitudes", lats1.length)
+      // create dim
+      val dims = new util.ArrayList[Dimension]()
+      val latDims = new util.ArrayList[Dimension]()
+      val lonDims = new util.ArrayList[Dimension]()
+      // add dims
+      dims.add(latDim)
+      dims.add(lonDim)
+      latDims.add(latDim)
+      lonDims.add(lonDim)
+      // add data
+      val rr = datafile.addVariable(null, "CE brightness temperature", DataType.INT, dims)
+      val latDimVar = datafile.addVariable(null, "lat", DataType.DOUBLE, latDims)
+      val lonDimVar = datafile.addVariable(null, "lon", DataType.DOUBLE, lonDims)
+      // add attributes
+      rr.addAttribute(new Attribute("units", "K"))
+      // add dim data
+      val latData = new ArrayDouble.D1(lats1.length)
+      val lonData = new ArrayDouble.D1(lons1.length)
+      for (i <- 0 until lats1.length) {
+        latData.set(i, lats1(i))
+      }
+      for (i <- 0 until lons1.length) {
+        lonData.set(i, lons1(i))
+      }
+      // create the file
+      datafile.create()
+      datafile.write(rr, varData)
+      datafile.write(latDimVar, latData)
+      datafile.write(lonDimVar, lonData)
+      datafile.close()
     }
+    catch {
+      case _: Throwable => logger.info("Error generating netCDF file for " + filepath + "\n")
+    }
+  }
+
 
+  /**
+   * Get the data from the nodes in the edge
+   *
+   * @param edge       the edge object which is composed of the nodes to look up in the MCSNodeMap
+   * @param MCSNodeMap mutable.HashMap[String, MCCNode] representing the map of each node metadata
+   */
+  def getMCSNodes(edge: MCCEdge, MCSNodeMap: mutable.HashMap[String, MCCNode]): (MCCNode, MCCNode) = {
```

Comment:
@sujen there's some design questions I had here.

1. Instead of returning a tuple of nodes - we could just return a populated edge object.

2. The node object in the mccedge seems to just be a shell and The MCSNodemap contains all the metadata on the nose.